lib/full_url_for.js

```diff
@@ -10,8 +10,8 @@ function fullUrlForHelper(path = '/') {
   const sitehost = parse(config.url).hostname || config.url;
   const data = new URL(path, `http://${sitehost}`);
 
-  // Exit if this is an external link
-  if (data.origin !== 'null' && data.hostname !== sitehost) return path;
+  // Exit if input is a data url or external link
+  if (data.origin === 'null' || data.hostname !== sitehost) return path;
```

Comment:
```suggestion
  if (data.hostname !== sitehost || data.origin === 'null') return path;
```

IMHO, external link should be more common than `mailto:`, `javascript:` url. If we change the order, we might increase performance by just a little bit.

Reply:
updated